cmd/access/unstaked_access_node_builder.go

```diff
@@ -132,3 +152,15 @@ func (builder *UnstakedAccessNodeBuilder) enqueueUnstakedNetworkInit(ctx context
 		return builder.UnstakedNetwork, err
 	})
 }
+
+// enqueueConnectWithStakedAN enqueues the upstream connector component which connects the libp2p host of the unstaked
+// AN with the staked AN.
+// Currently, there is an issue with LibP2P stopping advertisements of subscribed topics if no peers are connected
+// (https://github.com/libp2p/go-libp2p-pubsub/issues/442). This means that an unstaked AN could end up not being
+// discovered by other unstaked ANs if it subscribes to a topic before connecting to the staked AN. Hence, the need
+// of an explicit connect to the staked AN before the node attempts to subscribe to topics.
+func (builder *UnstakedAccessNodeBuilder) enqueueConnectWithStakedAN() {
+	builder.Component("unstaked network", func(_ cmd.NodeBuilder, _ *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+		return newUpstreamConnector(builder.stakedANIdentity, builder.UnstakedLibP2PNode, builder.Logger), nil
```

Comment:
Where is `builder.UnstakedLibP2PNode` defined? I didn't see it anywhere

Reply:
it is here https://github.com/onflow/flow-go/pull/1098/files#diff-9aaf6980a686547e7038272066b14b0e53809b3721d51162e25ecf2bdde00bd4R134